features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/ServiceCatalog/Listing/Delete.jsx

```diff
@@ -0,0 +1,111 @@
+import Alert from 'AppComponents/Shared/Alert';
+import ConfirmDialog from 'AppComponents/Shared/ConfirmDialog';
+import ServiceCatalog from 'AppData/ServiceCatalog';
+
+/**
+* Service Catalog service delete
+* @param {any} props Props for delete function.
+* @returns {any} Returns the rendered UI for service delete.
+*/
+function Delete(props) {
+    const intl = useIntl();
+    const [open, setOpen] = useState(false);
+    const toggleOpen = () => {
+        setOpen(!open);
+    };
+    const { serviceName, serviceId, getData } = props;
+
+    const deleteService = () => {
+        const deleteServicePromise = ServiceCatalog.deleteService(serviceId);
+        deleteServicePromise.then(() => {
+            Alert.info(intl.formatMessage({
+                id: 'ServiceCatalog.Listing.Delete.service.deleted.successfully',
+                defaultMessage: 'Service deleted successfully!',
+            }));
+            // Reload the services list
+            getData();
+        }).catch(() => {
+            Alert.error(intl.formatMessage({
+                defaultMessage: 'Error while deleting service',
+                id: 'ServiceCatalog.Listing.Delete.error.delete',
+            }));
+        });
+    };
+
+    const runAction = (confirm) => {
+        if (confirm) {
+            deleteService();
+        }
+        setOpen(!open);
+    };
+
+    return (
+        <>
+            <Button onClick={toggleOpen}>
+                <Icon>delete_forever</Icon>
+            </Button>
+            <ConfirmDialog
+                key='key-dialog'
+                labelCancel={(
+                    <FormattedMessage
+                        id='ServiceCatalog.Listing.Delete.cancel'
+                        defaultMessage='Cancel'
+                    />
+                )}
+                title={(
+                    <FormattedMessage
+                        id='ServiceCatalog.Listing.Delete.confirm'
+                        defaultMessage='Confirm Delete'
+                    />
+                )}
+                message={(
+                    <FormattedMessage
+                        id='ServiceCatalog.Listing.Delete.ok.confirm'
+                        defaultMessage='Are you sure you want to delete the service {service} ?'
+                        values={{ service: serviceName }}
+                    />
+                )}
+                labelOk={(
+                    <FormattedMessage
+                        id='ServiceCatalog.Listing.Delete.ok.yes'
+                        defaultMessage='Yes'
+                    />
+                )}
+                callback={runAction}
+                open={open}
+            />
+        </>
+    );
+}
+Delete.propTypes = {
+    classes: PropTypes.shape({}).isRequired,
+    serviceName: PropTypes.string.isRequired,
+    serviceId: PropTypes.string.isRequired,
+    getData: PropTypes.shape({}).isRequired,
+    intl: PropTypes.shape({}).isRequired,
+};
+
+export default injectIntl(Delete);
```

Comment:
Let's use `useIntl`  hook instead of this old style `injectIntl` HOC here 

```
import { useIntl } from 'react-intl';
```

Reply:
Fixed with https://github.com/wso2/carbon-apimgt/pull/9407/commits/a9d9f33a89aa8d840fac45a19758fa7270787f7c